e2e/app/page/notebook-page.ts

```diff
@@ -0,0 +1,185 @@
+import {ElementHandle, Frame, Page} from 'puppeteer';
+import {waitForDocumentTitle} from 'utils/waits-utils';
+import {getPropValue} from 'utils/element-utils';
+import AuthenticatedPage from './authenticated-page';
+import CodeCell, {Selector as CodeCellSelector} from './notebook-code-cell';
+
+// CSS selectors
+enum CssSelector {
+  body = 'body.notebook_app',
+  notebookContainer = '#notebook-container',
+  toolbarContainer = '#maintoolbar-container',
+  runCellButton = 'button[data-jupyter-action="jupyter-notebook:run-cell-and-select-next"]',
+  saveNotebookButton = 'button[data-jupyter-action="jupyter-notebook:save-notebook"]',
+  kernelIcon = '#kernel_indicator_icon',
+  kernelName = '.kernel_indicator_name',
+}
+
+enum Mode {
+  Command= 'command_mode',
+  Edit = 'command_mode',
+}
+
+export enum KernelStatus {
+  notRunning = 'Kernel is not running',
+  Idle = 'Kernel Idle',
+}
+
+export default class NotebookPage extends AuthenticatedPage {
```

Comment:
Can we add a test to make sure that on clicking any Download Option from File menu we see a pop Up with AoU policy

Reply:
I'll add this test in new PR next time.